test/Npgsql.Tests/Replication/CommonReplicationTests.cs

```diff
@@ -454,6 +454,43 @@ async Task<NpgsqlLogSequenceNumber> GetCommitLsn(string valueString)
 
         #endregion
 
+        #region BugTests
+
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/3534")]
+        public Task Bug3534()
+            => SafeReplicationTest(
+                async (slotName, tableName) =>
+                {
+                    await using var rc = await OpenReplicationConnectionAsync(new NpgsqlConnectionStringBuilder(ConnectionString) { CancellationTimeout = 1 });
+                    await CreateReplicationSlot(slotName);
+                    var info = await rc.IdentifySystem();
+                    using var streamingCts = new CancellationTokenSource();
```

Comment:
We can set the timeout in the constructor, instead of a `Task.Run`

Reply:
I don't really understand which constructor you mean, but I've improved the test according to @roji's suggestions.

Reply:
I don't really understand which constructor you mean, but I've improved the test according to @roji's suggestions.

Reply:
`CancellationTokenSource` has a constructor, which accepts the delay until it's cancelled.